### ISSUE TITLE
Track Designs Manager Crashing

### ISSUE THREAD

**Issue by Arystina** [OpenRCT2/OpenRCT2#4525]
**OS:** Windows 7
**Version:** 0.0.5
**Commit/Build:** 52e5506

I was editing tracks in the Track Designs Manager. I recently added a large pack of various player-made coaster tracks and wanted to organize them a bit better and after a few minutes the game crashed. I tried again, got back into the manager, edited two more tracks before the game crashed yet again. 

I am not sure what the issue is beyond that maybe I might have too many coaster tracks in my game files (roughly 3k). I thought I'd post here and find out for certain. 

Screenshot, save game (for whatever reason there is one), and dump file should all be in the zip.

[RCT2 TracksManager Crash.zip](https://github.com/OpenRCT2/OpenRCT2/files/507416/RCT2.TracksManager.Crash.zip)

EDIT: Removing tracks doesn't seem to make a difference with the crashing. Crash is now also affecting gameplay when scrolling through coasterbuild menus.

**Comment by Arystina** [comment]
Okay, so it's definitely something to do with the tracks. I did a completely fresh install of RCT2 using the latest OpenRCT2 build (71d0861) and ONLY moved in my tracks package, nothing else, (with some manual work I've dropped it from 3000 down to around 2500) and after a few minutes of browsing in-game through the tracks menu it did the same thing again with the crash.

**Comment by janisozaur** [comment]
```
 0  openrct2.dll!window_track_list_invalidate [track_list.c : 343 + 0x0]
    eip = 0x53d65e0c   esp = 0x000cfe38   ebp = 0x000cfedc   ebx = 0x54276508
    esi = 0x54276508   edi = 0x00000214   eax = 0x00000000   ecx = 0xffffffff
    edx = 0x5418a5f7   efl = 0x00210283
    Found by: given as instruction pointer in context
 1  openrct2.dll!window_find_widget_from_point [window.c : 891 + 0xc]
    eip = 0x53c66b38   esp = 0x000cfe40   ebp = 0x000cfedc
    Found by: call frame info
 2  openrct2.dll!game_handle_input_mouse [input.c : 286 + 0xa]
    eip = 0x53c39ef5   esp = 0x000cfe5c   ebp = 0x000cfedc
    Found by: call frame info
 3  openrct2.dll!game_handle_input [input.c : 143 + 0x2b]
    eip = 0x53c39bf1   esp = 0x000cfe7c   ebp = 0x000cfedc
    Found by: call frame info
 4  openrct2.dll!game_update [game.c : 340 + 0x5]
    eip = 0x53c37cac   esp = 0x000cfe90   ebp = 0x000cfedc
    Found by: call frame info
 5  openrct2.dll!rct2_update [rct2.c : 457 + 0x5]
    eip = 0x53c3e9b5   esp = 0x000cfea0   ebp = 0x000cfedc
    Found by: call frame info
 6  openrct2.dll!openrct2_loop [openrct2.c : 466 + 0x5]
    eip = 0x53c3dbe1   esp = 0x000cfeb0   ebp = 0x000cfedc
    Found by: call frame info
 7  openrct2.dll!openrct2_launch [openrct2.c : 338 + 0x5]
    eip = 0x53c3d744   esp = 0x000cfee4   ebp = 0x000cff88
    Found by: call frame info
 8  openrct2.dll!StartOpenRCT [windows.c : 131 + 0x5]
    eip = 0x53cab387   esp = 0x000cfee8   ebp = 0x000cff88
    Found by: call frame info
 9  openrct2.exe + 0x69e4
    eip = 0x004069e4   esp = 0x000cfef0   ebp = 0x000cff88
    Found by: call frame info
10  kernel32.dll + 0x1338a
    eip = 0x7634338a   esp = 0x000cff90   ebp = 0x000cff94
    Found by: previous frame's frame pointer
11  ntdll.dll + 0x39882
    eip = 0x779d9882   esp = 0x000cff9c   ebp = 0x000cffd4
    Found by: previous frame's frame pointer
12  ntdll.dll + 0x39855
    eip = 0x779d9855   esp = 0x000cffdc   ebp = 0x000cffec
    Found by: previous frame's frame pointer
```

**Comment by Arystina** [comment]
Okay, what's that mean?

**Comment by IntelOrca** [comment]
Do you only get a crash in the track manager?

**Comment by Arystina** [comment]
No, I get it in game as well when browsing through premade coasters to pick one to build. I first noticed it in the track manager though.

How do I select by track type? I'm looking but not seeing that anywhere.

**Comment by Arystina** [comment]
A number of the coasters I have installed are missing custom scenery which I no longer have and am unable to get (the site that I got it from is long dead). Could that be causing the crashes or does missing scenery not impact that? That's really the only thing I can think of at this point other than there being a limit on the number of tracks we can have in the tracks folder.

**Comment by IntelOrca** [comment]
The crash dump I looked at showed that it was trying to access a ride entry that wasn't loaded for that park.

**Comment by Arystina** [comment]
All I did was scroll through and rename/delete coasters I didn't want in the tracks manager, and then later tried playing a game. -is confused- Not sure why the tracks are causing this much trouble.

**Comment by IntelOrca** [comment]
I think it will be easier if you zip all your tracks and upload them - then I will see what's going on.

**Comment by Arystina** [comment]
Sorry, I just saw your message. Fortunately, I don't think I need to send you anything. With your help, I was able to figure out the problem. You said it looked like it was trying to access a ride entry that wasn't loaded for the park. It was indeed. Turns out the game crashed always on particular ride names, and on screen, I noticed those rides did not have an image show up. I managed to look a few up, and it turns out those rides have modded vehicles, which I do not have added in the game, thus resulting in the crashes. It will be time consuming as I have to bear with the crashes and go through each ride at a time, but eventually I should be able to get rid of all the modded ones and resolve the crashes. I do apologize for having taken up your time for my user-error.

**Comment by Arystina** [comment]
*cries* Now I can't even open up the custom scenarios. I get a different error message. This has kept me up all night. I give up. Here.

[Ary's Broken Stuff.zip](https://github.com/OpenRCT2/OpenRCT2/files/511522/Ary.s.Broken.Stuff.zip)

I tried zipping my entire game to you but apparently it's too big so I've opted for two of the scenario packages and their error messages + crash report. Build is still 71d0861.

What a mess. Something hates me.

**Comment by IntelOrca** [comment]
Regarding the CalculateExtraBytesToFixChecksum error:

Some custom objects were saved with incorrect checksums, we therefore try to fix it without changing the checksum by appending a clever set of bytes on the end of the object. But there are cases where this fails to get the right checksum. Its not a big deal as it shouldn't stop anything from working other than give a few warnings when it generates the object cache.

Click ignore when you see the assertion message and it should work fine.

As for the tracks

Can you send me some that crash the game (have "modded" vehicles) so that I can prevent the crash?

**Comment by Arystina** [comment]
Okay, thanks. Works now. I thought I was out of luck with those scenarios.

Seems I still get crashes even after removing the modded rides. Happens after deleting a series of rides in a row in the manager. Takes a few minutes of browsing, editing, and deleting. I will just give up and go ahead and send you all the track files (this will include the game's coasters as well since I am too tired to sort them out from the player ones) and a few DAT files that I happened to have that are custom vehicles. I will include a separate zip of a crash of a missing vehicle. To duplicate for yourself, remove the AE-WSLID.DAT file that I included from your game ObjData folder and go into the tracks manager and go to water dinghies and scroll to basic waterslide or grapevine. It should crash like it did for me, as these slides require that file.
[Ary's Tracks File.zip](https://github.com/OpenRCT2/OpenRCT2/files/511833/Ary.s.Tracks.File.zip)
[Modded Vehicle Missing Crash Report.zip](https://github.com/OpenRCT2/OpenRCT2/files/511834/Modded.Vehicle.Missing.Crash.Report.zip)
